### ISSUE TITLE
Could not find a part of the path with CS-Script 4.0.0.0

### ISSUE THREAD

**Issue by UweKeim** [oleg-shilo/cs-script#234]
Just gave version 4.0.0.0 a try by running a _simple_ script.

As stated in the [release notes](https://github.com/oleg-shilo/cs-script/releases/tag/v4.0.0.0) I've simply unpacked the downloaded 7z file and try to run it.

I get an error:

> Could not find a part of the path 'C:\Users\MyUser\AppData\Local\Temp\csscript.core\build\obj'.

This is the batch file that I try to run:

```cmd
PUSHD 
CD /d %~dp0 

SET CSSCRIPT_DIR=%~dp0\cs-script

"%~dp0\cs-script\cscs.exe" /dbg "%~dp0\run.cs" %*

POPD 
EXIT /B 0
```

And this is the complete cs-script file:

```csharp
using System:

public static class Install
{
	public static int Main(string[] args)
	{
		log("Started."):
		
		// TODO.

		log("Finished."):
		return 0;
	}
	
	private static void log(string text)
	{
		Console.WriteLine(text):
	}
}
```

My folder structure is:

* `C:\MyFolder\MyScript` (Contains both my .cmd file and my "run.cs" file)
* `C:\MyFolder\MyScript\cs-script` (Contains the extracted 7z file)

When calling

    dotnet --list-runtimes

I get:

```text
Microsoft.AspNetCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
```

### My question:

Am I doing something wrong?

**Comment by oleg-shilo** [comment]
Hi Uwe, 
will check

**Comment by oleg-shilo** [comment]
Not sure what is happening.

Your test-case works OK for me. Just to avoid any discrepancy I have attached the working test case (very same that you have shared)

[support.zip](https://github.com/oleg-shilo/cs-script/files/6162004/support.zip)

Though I have slightly different runtimes installed:
```
D:\dev\support>dotnet --list-runtimes
Microsoft.AspNetCore.All 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
Microsoft.AspNetCore.App 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 2.1.24 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 3.1.10 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 3.1.11 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.2 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```
It looks like I am running a more complete .NETCore (e.g. with desktop)

Can you please check the attached test case to ensure we are running the same things

**Comment by UweKeim** [comment]
Thanks a lot, Oleg, running your test case brings the same error:

```
c:\Ablage\support>c:\Ablage\support\run.cmd

PUSHD

CD /d c:\Ablage\support\

SET CSSCRIPT_DIR=c:\Ablage\support\\cs-script

"c:\Ablage\support\\cs-script\cscs.exe" -c:0 /dbg "c:\Ablage\support\\run.cs"
Error: Specified file could not be compiled.

Could not find a part of the path 'C:\Users\UweKeim\AppData\Local\Temp\csscript.core\build\obj'.

POPD

EXIT /B 0
```

I've also updated the used "HostingBundle" to 5.0.4 before.

Next I've installed the Desktop runtime:

```
C:\>dotnet --list-runtimes
Microsoft.AspNetCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 3.1.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.WindowsDesktop.App 5.0.4 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
```

Still, the same error occurs.

------

Just to clarify, as I should have mentioned this earlier, this is a **Windows Server 2016** Standard, Version 1607, 64-Bit.

When trying to run it on my dev machine (Windows 10, 64-bit, German), the script runs successfully.

I've tried several other computers and it seems that it runs only on computers that have some development environment installed. I'm assuming that the .NET Core SDK is required.

Because I found another Windows Server where it succeeded (.NET Core SDK installed) and yet another Windows 10 PC where it did _not_ succeed (no .NET Core SDK installed).

Could be a red herring, but I do think that you should be able to see the error if you use the "support.zip" inside a fresh new Windows 10 virtual machine.

**Comment by oleg-shilo** [comment]
This is an excellent clue.
Yes  SDK is a dependency.
On Choco .NET 5 SDK is a dependency of the package so it is installed automatically. 

I am still behind with the documentation and will need to reflect the need for SDK explicitly.

Though... I would still expect .NET Runtime to be able to run.
Can you please execute `cscs.exe`? It will print the found compilers. I suspect it cannot find them in your case.

**Comment by UweKeim** [comment]
```
c:\Ablage\support\cs-script>cscs
C# Script execution engine (.NET Core). Version 4.0.0.0.
Copyright (C) 2004-2020 Oleg Shilo. www.csscript.net (github.com/oleg-shilo/cs-script.core)

   CLR:             5.0.4
   System:          Microsoft Windows NT 10.0.14393.0
   Architecture:    x64
   Install dir:     <not integrated>
   Script engine:   c:\Ablage\support\cs-script\cscs.dll
   Config file:     <none>
   Compiler engine: dotnet (C:\Program Files\dotnet\dotnet.exe)
   NuGet manager:   dotnet
   NuGet cache:     <not found>
   Custom commands: C:\ProgramData\cs-script\commands
   Global includes: C:\ProgramData\cs-script\inc
```

**Comment by oleg-shilo** [comment]
Hm...
It found it. The dotnet executable.

Can you now try to run your script with an extra CLI arg `-ng:` to ensure dotnet but not csc is used.

In fact try both:

```ps
cscs -ng:csc -c:0 test.cs 
cscs -ng:dotnet -c:0 test.cs 
```

**Comment by UweKeim** [comment]
Here is the result:

```
c:\Ablage\support\cs-script>cscs -ng:csc -c:0 c:\Ablage\support\run.cs
Error: Specified file could not be compiled.

c:\Ablage\support\run.cs(0,0): error CS0000:
Build server error: System.ArgumentNullException: Value cannot be null. (Parameter 'assemblyFile')
   at System.AppDomain.ExecuteAssembly(String assemblyFile, String[] args)
   at CSScripting.CodeDom.BuildServer.Compile(String compiler, String[] args)


c:\Ablage\support\cs-script>cscs -ng:dotnet -c:0 c:\Ablage\support\run.cs
Error: Specified file could not be compiled.

Could not find a part of the path 'C:\Users\KeimUwe\AppData\Local\Temp\csscript.core\build\obj'.
```

**Comment by oleg-shilo** [comment]
I have prepared RC for v4.0.1.0.
It supports a new switch `-verbose2` that prints the compiling command and the compiler output.
The intention is to help in the cases like yours, when I cannot easily access the required environment to reproduce the problem.
I have attached the binaries.

![image](https://user-images.githubusercontent.com/16729806/111628188-0b14d380-8844-11eb-9d97-9862a7f1d96e.png)

Please give it a try and see what exactly is upsetting the compiler

[cs-script.win.v4.0.1.0.7z.zip](https://github.com/oleg-shilo/cs-script/files/6164144/cs-script.win.v4.0.1.0.7z.zip)

**Comment by UweKeim** [comment]
Thanks, Oleg,

here ich the output of your support-script with the new version:

```
C:\Ablage\support>"C:\Ablage\support\run.cmd"

PUSHD

CD /d C:\Ablage\support\

SET CSSCRIPT_DIR=C:\Ablage\support\\cs-script

"C:\Ablage\support\\cs-script\cscs.exe" -verbose2 -c:0 /dbg "C:\Ablage\support\\run.cs"
> ----------------
  TragetFramework: v4.0
  Provider:
  Script engine: C:\Ablage\support\cs-script\cscs.dll
  Compiler engine: dotnet (C:\Program Files\dotnet\dotnet.exe)
  Console Encoding: Codepage - 850 (Codepage - 850) - system default
  CurrentDirectory: C:\Ablage\support
  NuGet manager: dotnet
  NuGet cache: <not found>
  Executing: C:\Ablage\support\run.cs
  Script arguments:
  SearchDirectories:
    ------- (local dirs) -------
    0 - C:\Ablage\support
    1 - C:\Ablage\support\cs-script
    ------- (dirs from cmd args) -------
    ------- (dirs from code) -------
    ------- (dirs from config) -------
    2 - C:\Ablage\support\cs-script\lib
    3 - C:\ProgramData\cs-script\commands
    4 - C:\ProgramData\cs-script\inc
    ------- (cs-script special dirs) -------
    5 - C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\cache\-1609942760
> ----------------

Compiling script...

  Precompilers:
   0 - run.cs -> csscript.DefaultPrecompiler
           from C:\Ablage\support\cs-script\cscs.dll

  Precompilers:
   0 - run.cs -> csscript.AutoclassPrecompiler
           from C:\Ablage\support\cs-script\cscs.dll

  Output file:
       C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\13704.a0d1d7c5-9f77-4759-ab6e-74409bbf80fd.dll

  Files to compile:
   0 - C:\Ablage\support\\run.cs
   1 - C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\cache\-1609942760\run.attr.g.cs

  References:
   0 - System
   1 - System.Core
> ----------------
Error: Specified file could not be compiled.

Could not find a part of the path 'C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\build\obj'.

POPD

EXIT /B 0
```

**Comment by oleg-shilo** [comment]
I found the trigger for the problem. In absence of SDK a C# project template cannot be created so dotnet.exe cannot use it to compile the script to assembly.
csc.dll can do the job but it is not available if sdk is not installed.

I am yet to see why we are failing to create a template but let's try this:
- investigation
  Try to execute `dotnet new console -lang C#` and let me know the produced folder structure
- potential temp work around
  In `C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\build` drop the attached project template file manually.
  let me know if you can run the scripts now.
  
[uwe.zip](https://github.com/oleg-shilo/cs-script/files/6168934/uwe.zip)

**Comment by UweKeim** [comment]
Investigation results:

```
C:\Ablage\oleg1>dotnet new console -lang C#
Could not execute because the application was not found or a compatible .NET SDK is not installed.
Possible reasons for this include:
  * You intended to execute a .NET program:
      The application 'new' does not exist.
  * You intended to execute a .NET SDK command:
      It was not possible to find any installed .NET SDKs.
      Install a .NET SDK from:
        https://aka.ms/dotnet-download
```

Workaround results (after dropping the "uwe.csproj" in `C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\build`):

```
C:\Ablage\oleg1>"C:\Ablage\support\run.cmd"

PUSHD

CD /d C:\Ablage\support\

SET CSSCRIPT_DIR=C:\Ablage\support\\cs-script

"C:\Ablage\support\\cs-script\cscs.exe" -verbose2 -c:0 /dbg "C:\Ablage\support\\run.cs"
> ----------------
  TragetFramework: v4.0
  Provider:
  Script engine: C:\Ablage\support\cs-script\cscs.dll
  Compiler engine: dotnet (C:\Program Files\dotnet\dotnet.exe)
  Console Encoding: Codepage - 850 (Codepage - 850) - system default
  CurrentDirectory: C:\Ablage\support
  NuGet manager: dotnet
  NuGet cache: <not found>
  Executing: C:\Ablage\support\run.cs
  Script arguments:
  SearchDirectories:
    ------- (local dirs) -------
    0 - C:\Ablage\support
    1 - C:\Ablage\support\cs-script
    ------- (dirs from cmd args) -------
    ------- (dirs from code) -------
    ------- (dirs from config) -------
    2 - C:\Ablage\support\cs-script\lib
    3 - C:\ProgramData\cs-script\commands
    4 - C:\ProgramData\cs-script\inc
    ------- (cs-script special dirs) -------
    5 - C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\cache\-1609942760
> ----------------

Compiling script...

  Precompilers:
   0 - run.cs -> csscript.DefaultPrecompiler
           from C:\Ablage\support\cs-script\cscs.dll

  Precompilers:
   0 - run.cs -> csscript.AutoclassPrecompiler
           from C:\Ablage\support\cs-script\cscs.dll

  Output file:
       C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\3044.bd2f6a5c-5b58-466c-a21d-1a7c5d30ba8a.dll

  Files to compile:
   0 - C:\Ablage\support\\run.cs
   1 - C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\cache\-1609942760\run.attr.g.cs

  References:
   0 - System
   1 - System.Core
> ----------------
Error: Specified file could not be compiled.

Could not find a part of the path 'C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\build\obj'.

POPD

EXIT /B 0
```

Sorry for all the trouble you are having with my issue! (I'm urgently trying to avoid installing an SDK on all my production servers)

**Comment by oleg-shilo** [comment]
OK, I think I may have found the problem. The actual problem.
The built-in fall back mechanism that I implemented for no-sdk scenario was crippled by an early mistake in the fallback trigger.

I have fixed it. Try the attached build v4.0.1. It will create the project file reliably.

We may still fail if dotnet.exe cannot build the assembly but we will test it too.

Uwe, this is what you need to try:
1. Remove the  csproj file you created manually.
2. Try your test with the attahhed 4.0.1.1 build.
3. If it succeeds go and celebrate it with some beer. :) 
  If not then try the attached test.zip file: execute test.cmd. It will check if an assembly can be built at all.
  
 There is a good chance you will not need to do step 3.
[test.zip](https://github.com/oleg-shilo/cs-script/files/6169966/test.zip)

[cs-script.win.v4.0.1.1.7z.zip](https://github.com/oleg-shilo/cs-script/files/6169862/cs-script.win.v4.0.1.1.7z.zip)

**Comment by UweKeim** [comment]
Okay, I've tested "test.zip" with these results:

```
C:\Ablage\support>c:\Ablage\Test\test.cmd

dotnet build --configuration Release -o bin
Could not execute because the application was not found or a compatible .NET SDK is not installed.
Possible reasons for this include:
  * You intended to execute a .NET program:
      The application 'build' does not exist.
  * You intended to execute a .NET SDK command:
      It was not possible to find any installed .NET SDKs.
      Install a .NET SDK from:
        https://aka.ms/dotnet-download

pause
```

When running my "run.cmd" that still looks like that:

```cs
PUSHD 
CD /d %~dp0 

SET CSSCRIPT_DIR=%~dp0\cs-script

"%~dp0\cs-script\cscs.exe" -verbose2 -c:0 /dbg "%~dp0\run.cs" %*

POPD 
EXIT /B 0
```

I'll get:

```
C:\Ablage\support>"C:\Ablage\support\run.cmd"

PUSHD

CD /d C:\Ablage\support\

SET CSSCRIPT_DIR=C:\Ablage\support\\cs-script

"C:\Ablage\support\\cs-script\cscs.exe" -verbose2 -c:0 /dbg "C:\Ablage\support\\run.cs"
> ----------------
  TragetFramework: v4.0
  Provider:
  Script engine: C:\Ablage\support\cs-script\cscs.dll
  Compiler engine: dotnet (C:\Program Files\dotnet\dotnet.exe)
  Console Encoding: Codepage - 850 (Codepage - 850) - system default
  CurrentDirectory: C:\Ablage\support
  NuGet manager: dotnet
  NuGet cache: <not found>
  Executing: C:\Ablage\support\run.cs
  Script arguments:
  SearchDirectories:
    ------- (local dirs) -------
    0 - C:\Ablage\support
    1 - C:\Ablage\support\cs-script
    ------- (dirs from cmd args) -------
    ------- (dirs from code) -------
    ------- (dirs from config) -------
    2 - C:\Ablage\support\cs-script\lib
    3 - C:\ProgramData\cs-script\commands
    4 - C:\ProgramData\cs-script\inc
    ------- (cs-script special dirs) -------
    5 - C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\cache\-1609942760
> ----------------

Compiling script...

  Precompilers:
   0 - run.cs -> csscript.DefaultPrecompiler
           from C:\Ablage\support\cs-script\cscs.dll

  Precompilers:
   0 - run.cs -> csscript.AutoclassPrecompiler
           from C:\Ablage\support\cs-script\cscs.dll

  Output file:
       C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\8720.fa3997f9-ac55-4566-8655-bfac2878e3df.dll

  Files to compile:
   0 - C:\Ablage\support\\run.cs
   1 - C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\cache\-1609942760\run.attr.g.cs

  References:
   0 - System
   1 - System.Core
> ----------------
error> Could not execute because the application was not found or a compatible .NET SDK is not installed.
error> Possible reasons for this include:
error>   * You intended to execute a .NET program:
> ================
dotnet.exe run:
  current_dir: C:\Users\KeimUwe\AppData\Local\Temp\2\csscript.core\cache\-1609942760\.build\run.cs
  cmd: dotnet build --configuration Debug -o bin
error>       The application 'build' does not exist.
error>   * You intended to execute a .NET SDK command:
error>       It was not possible to find any installed .NET SDKs.
error>       Install a .NET SDK from:
error>         https://aka.ms/dotnet-download
  output:

> ================
Error: Specified file could not be compiled.

BUILD: error : Unknown compiler error


POPD

EXIT /B 0
```

I do think I've used the correct version of CS-Script, although I cannot find any "4.0.1.1.7" written somewhere.

```
C:\Ablage\support>"C:\Ablage\support\cs-script\cscs.exe"
C# Script execution engine (.NET Core). Version 4.0.1.1.
Copyright (C) 2004-2020 Oleg Shilo. www.csscript.net (github.com/oleg-shilo/cs-script.core)

   CLR:             5.0.4
   System:          Microsoft Windows NT 10.0.14393.0
   Architecture:    x64
   Install dir:     <not integrated>
   Script engine:   C:\Ablage\support\cs-script\cscs.dll
   Config file:     <none>
   Compiler engine: dotnet (C:\Program Files\dotnet\dotnet.exe)
   NuGet manager:   dotnet
   NuGet cache:     <not found>
   Custom commands: C:\ProgramData\cs-script\commands
   Global includes: C:\ProgramData\cs-script\inc
```

**Comment by UweKeim** [comment]
When reading [the documentation for `dotnet build`](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build), there is a remark written:

> This article applies to: ✔️ .NET Core 2.x SDK and later versions

So from my understanding there is no possible way of using `dotnet build` _without_ having the SDK installed?!?

**Comment by oleg-shilo** [comment]
>4.0.1.1.7

:) it was a file name of the zip file. The release is automatically packed as *.7z but GitHub requires it to be re-packed as zip so it became *.7z.zip.
![image](https://user-images.githubusercontent.com/16729806/111760487-e166b580-88f2-11eb-878f-0212029dbc1e.png)


Uwe, we have done everything we could. We just need the SDK to be present :(

The only last chance (very slim one) is to see if .NET runtime still deploys csc.exe but in the location that I did not anticipate. If it does then we can still try to use is as a default compiler. But probably it is not included :(

Can you please check it?

**Comment by UweKeim** [comment]
OMG, shame on me. Of course it is the "7" from the 7-zip file extension. Doh!

![image](https://user-images.githubusercontent.com/415439/111763767-ca987280-88a2-11eb-8026-c28fa56b58bd.png)

I've installed the SDK now on the machines where I want to try CS-Script with .NET 5.

Thanks for your awesome support, Oleg.

**Comment by oleg-shilo** [comment]
TBH I did not fully realized the depth of the SDK-dependency.
Thank you, you have triggered this useful investigation. Now I also know that my no-sdk fallback routing helps with the creating the script execution infrastructure but... does nothing for achieving the desired outcome 😞   

Still, it's a usefull learning.

Thank you Uwe. As usual our interactions are mutually beneficial.
Have a good one.

**Comment by UweKeim** [comment]
Maybe better/more error messages like e.g. 

> ".NET Core SDK not found but required. Please install before using CS-Script."

or something similar could help other users to not step into the same trap as I did.

**Comment by oleg-shilo** [comment]
Yep, absolutely. 
Now when we have proved that it is not desirable but essential the startup check is a must.

**Comment by oleg-shilo** [comment]
Hey Uwe, last ask :)

Can you check please what is the output of the `dotnet --list-sdks` on the PC without SDK installed.

I am putting the checking at startup

**Comment by UweKeim** [comment]
In my tests:

```
C:\Windows\system32>dotnet --list-sdks

```

I. e. it returns nothing/empty string/blanks (can't say for sure).

In contrast, on the same machine:

```
C:\Windows\system32>dotnet --list-runtimes
Microsoft.AspNetCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
Microsoft.NETCore.App 3.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
Microsoft.NETCore.App 5.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
```

**Comment by oleg-shilo** [comment]
Great, thank you. 
Then I will use it as an SDK test

**Comment by oleg-shilo** [comment]
This issue has been closed as a more informative error handling notification (on no .NET SDK found) has been implemented.

However, I just realized that the old Roslyn based compilation engine (POC I did before .NET5 was releases) can be used to 
to execute scripts without SDK.

This can be a solution for the the cases when the low dependency is the primary requirement and the loss of some flexibility is acceptable.  

I have resurrected the POC and it seems to work. The feature will be available with both CLI params and in-code directives (similar to `dotnet` and `csc`):

```ps
css -ng:roslyn test.cs
``` 
```c#
//css_ng roslyn
using System;
. . .
``` 

I am not yet declaring it done 😄 but it looks like I will be able to do it.
